libs/pilight/protocols/GPIO/bme280.c

```diff
@@ -0,0 +1,540 @@
+	struct JsonNode *json = (struct JsonNode *) node->param;
+	struct JsonNode *jid = NULL;
+	struct JsonNode *jchild = NULL;
+	struct settings_t *bme280data = MALLOC(sizeof(struct settings_t));
+	int y = 0, interval = 10, nrloops = 0;
+	char *stmp = NULL;
+	double itmp = -1, temp_offset = 0, pressure_offset = 0;
+	uint8_t hum_oversampling = 1;
+        uint8_t tem_oversampling = 1;
+        uint8_t pre_oversampling = 1;
+
+	if(bme280data == NULL) {
+		fprintf(stderr, "out of memory\n");
+		exit(EXIT_FAILURE);
+	}
+
+	bme280data->nrid = 0;
+	bme280data->id = NULL;
+	bme280data->fd = 0;
+	bme280data->dig_t1 = 0;
+	bme280data->dig_t2 = 0;
+	bme280data->dig_t3 = 0;
+	bme280data->dig_p1 = 0;
+	bme280data->dig_p2 = 0;
+	bme280data->dig_p3 = 0;
+        bme280data->dig_p4 = 0;
+        bme280data->dig_p5 = 0;
+        bme280data->dig_p6 = 0;
+        bme280data->dig_p7 = 0;
+        bme280data->dig_p8 = 0;
+        bme280data->dig_p9 = 0;
+	bme280data->dig_h1 = 0;
+	bme280data->dig_h2 = 0;
+        bme280data->dig_h3 = 0;
+        bme280data->dig_h4 = 0;
+        bme280data->dig_h5 = 0;
+        bme280data->dig_h6 = 0;
+
+	threads++;
+
+	if((jid = json_find_member(json, "id"))) {
+		jchild = json_first_child(jid);
+		while(jchild) {
+			if(json_find_string(jchild, "id", &stmp) == 0) {
+				if((bme280data->id = REALLOC(bme280data->id, (sizeof(char *) * (size_t)(bme280data->nrid + 1)))) == NULL) {
+					fprintf(stderr, "out of memory\n");
+					exit(EXIT_FAILURE);
+				}
+				if((bme280data->id[bme280data->nrid] = MALLOC(strlen(stmp) + 1)) == NULL) {
+					fprintf(stderr, "out of memory\n");
+					exit(EXIT_FAILURE);
+				}
+				strcpy(bme280data->id[bme280data->nrid], stmp);
+				bme280data->nrid++;
+			}
+			if(json_find_string(jchild, "i2c-path", &stmp) == 0) {
+				strcpy(bme280data->path, stmp);
+			}
+			jchild = jchild->next;
+		}
+	}
+
+	if(json_find_number(json, "poll-interval", &itmp) == 0)
+		interval = (int) round(itmp);
+	json_find_number(json, "temperature-offset", &temp_offset);
+	json_find_number(json, "pressure-offset", &pressure_offset);
+	if(json_find_number(json, "hum-oversampling", &itmp) == 0) {
+		hum_oversampling = (uint8_t) itmp;
+	}
+	hum_oversampling &= 0x7;
+        if(json_find_number(json, "tem-oversampling", &itmp) == 0) {
+                tem_oversampling = (uint8_t) itmp;
+        }
+        tem_oversampling &= 0x7;
+
+        if(json_find_number(json, "pre-oversampling", &itmp) == 0) {
+                pre_oversampling = (uint8_t) itmp;
+        }
+        pre_oversampling &= 0x7;
+
+	// resize the memory blocks pointed to by the different pointers
+	size_t sz = (size_t) (bme280data->nrid + 1);
+	unsigned long int sizeUChar = sizeof(uint8_t) * sz;
+        unsigned long int sizeChar = sizeof(int8_t) * sz;
+	unsigned long int sizeShort = sizeof(int32_t) * sz;
+	unsigned long int sizeUShort = sizeof(uint32_t) * sz;
+	bme280data->fd = REALLOC(bme280data->fd, (sizeof(int) * sz));
```

Comment:
Can you error check if these succeeded?

Reply:
Isn't that done in the following ? :
`        bme280data->dig_h6 = REALLOC(bme280data->dig_h6, sizeChar);

        if(bme280data->dig_t1 == NULL || bme280data->dig_t2 == NULL || bme280data->dig_t3 == NULL || bme280data->dig_p1 == NULL ||
                bme280data->dig_p2 == NULL || bme280data->dig_p3 == NULL || bme280data->dig_p4 == NULL || bme280data->dig_p5 == NULL $
                bme280data->dig_p6 == NULL || bme280data->dig_p7 == NULL || bme280data->dig_p8 == NULL || bme280data->dig_p9 == NULL $
                bme280data->dig_h1 == NULL || bme280data->dig_h2 == NULL || bme280data->dig_h3 == NULL || bme280data->dig_h4 == NULL $
                bme280data->dig_h5 == NULL || bme280data->dig_h6 == NULL || bme280data->fd == NULL) {
                fprintf(stderr, "out of memory\n");
                exit(EXIT_FAILURE);
        }
`

Reply:
Yes, missed that.